### ISSUE TITLE
Custom enum and enum-method-mappings

### ISSUE THREAD

**Issue by jenscski** [xamarin/xamarin-android#3869]
### Steps to Reproduce

1. Defined an enum in Additions
2. Create mapping enum-method-mappings
3. The enum is not found

### Log File

BINDINGSGENERATOR : warning BG8700: Unknown return type Bluebird.Enum.SDCommonResult in method SD_Connect in managed type Bluebird.Sled.ISerialManager.
BINDINGSGENERATOR : warning BG8503: Invalidating Bluebird.Sled.ISerialManager and all nested types because some of its methods were invalid.
BINDINGSGENERATOR : warning BG8C00: For type Bluebird.Sled.Reader, base interface co.kr.bluebird.sled.ISerialManager is invalid.
BINDINGSGENERATOR : warning BG8C00: For type Bluebird.Sled.Reader, base interface co.kr.bluebird.sled.ISerialManager is invalid.

**Comment by jpobst** [comment]
Unfortunately `generator` runs before the C# compiler and doesn't know about anything you add in `Additions`.  You will need to use the `EnumFields.xml` file to create your enum.  You will need to put valid-formed JNI information in it, but it doesn't need to point to actual classes/fields.

Example:
```
<mapping jni-class="blah/blah" clr-enum-type="Bluebird.Enum.SDCommonResult">
  <field jni-name="name" clr-name="Name" value="0" />
  <field jni-name="id" clr-name="Id" value="1" />
  <field jni-name="tag" clr-name="Tag" value="2" />
</mapping>
```

**Comment by jenscski** [comment]
Yes, that would work, but it would create a lot of false warnings like, so don't think I will go that way.


```
warning BG8A00: <remove-node path="/api/package[@name='blah']/class[@name='blah']/field[@name='name']"/> matched no nodes.
warning BG8A00: <remove-node path="/api/package[@name='blah']/class[@name='blah']/field[@name='id']"/> matched no nodes.
warning BG8A00: <remove-node path="/api/package[@name='blah']/class[@name='blah']/field[@name='tag']"/> matched no nodes.

```
